stdlib/cmake/modules/SwiftSource.cmake

```diff
@@ -656,15 +656,15 @@ function(_compile_swift_files
            "-emit-private-module-interface-path" "${private_interface_file}")
     endif()
 
-    if (NOT SWIFTFILE_IS_STDLIB_CORE)
+    if(SWIFT_STDLIB_EMIT_API_DESCRIPTORS AND NOT SWIFTFILE_IS_FRAGILE)
```

Comment:
To make the diff more reviewable, I should likely put this after `if (NOT SWIFTFILE_IS_STDLIB_CORE)`

Reply:
I think it makes more sense to defer the removal of the redundant code -- there is more value in keeping the code handling the emission of the api descriptors near the one handling the emission of the swift interfaces, so that it "matches" the section doing the same for macCatalyst.